test/e2e/network/ingressclass.go

```diff
@@ -73,7 +73,7 @@ var _ = common.SIGDescribe("IngressClass [Feature:Ingress]", func() {
 		}
 	})
 
-	ginkgo.It("should prevent Ingress creation if more than 1 IngressClass marked as default [Serial]", func() {
+	ginkgo.It("should choose the one with the lower name when two ingressClasses are marked as default and have the same CreationTimestamp [Serial]", func() {
```

Comment:
I don't think this will have the same timestamp, infress classes are created sequentially ... if we want to be completely sure just add a time.Sleep(100*time.Millisecond) between both creations

Reply:
Or maybe we can calcluate for the expected ingressClass before creating tht ingress?
```
expectedName := ingressClass1.Name
if ingressClass2.CreationTimestamp.UnixNano() > ingressClass1.CreationTimestamp.UnixNano() {
	expectedName = ingressClass2.Name
}
```
And compare ingress.className with `expectedName` int the test?

Reply:
I found that `the admission controller may take a few seconds to observe both ingress classes`. Maybe we should retest until the timeout is reached instead of return an error when ingress.className is not equal to `expectedName`?